### PR TITLE
Added GUI version of Drifty

### LINKED ISSUE #4: [FEATURE] Making a gui version of Drifty
**Is your feature request related to a problem? Please describe.**
It's not always loved by users to use console applications, rather they find much interest in using GUI (Graphical User Interface) version of any application.

**Describe the solution you'd like**
We nees to build a GUI version of the application. We can use JavaFX or swing for that purpose.

**Additional context**
Contributors who want to take over this issue should at first comment below and then, I would assign them with the task. Multiple assignes can be there.

### PR DESCRIPTION
**Pull request by SaptarshiSarkar12**
Closes #4
---
# Changes Proposed -
- Separated `Backend` from CLI and added GUI functionalities.
- Fixed YouTube video not downloading when yt-dlp is in temporary directory.
- Changed the project to a Maven project.
- Simplified Progress bar for CLI for multithreaded downloading of a file.

# Changes that needs help -
- YouTube video download progress is not show in the progress bar of the GUI.

### CODE REVIEW

**Review by JaiS99 — APPROVED** [review]
Approved!

**Comment by SaptarshiSarkar12** [comment]
Thank you @JaiS99 for your review :+1: !

File: src/main/java/Utils/Constants.java
Comment (OkelloSam21) [inline src/main/java/Utils/Constants.java:1]: This Is okay.... but when I pasted it I got the following for you
Gitignore suggestions  - Google Chrome 3_26_2023 2_26_34 AM (2)

**Review by OkelloSam21 — APPROVED** [review]
I couldn't view all files but I like the way you write your code. It's cool.

**Comment by SaptarshiSarkar12** [comment]
okay @OkelloSam21 . Thank you for your reviews.

File: src/main/java/Utils/Constants.java
Comment (SaptarshiSarkar12) [reply src/main/java/Utils/Constants.java:1]: ok, I shall make those changes. Thank you @OkelloSam21 :+1:

**Comment by SaptarshiSarkar12** [comment]
> I couldn't view all files but I like the way you write your code. It's cool.

@OkelloSam21 It's okay Samuel, thank you for your appreciation :heart: !
I am glad you liked my code!

**Review by OkelloSam21 — APPROVED** [review]

**Review by SaptarshiSarkar12 — COMMENTED** [review]
LGTM :+1:

**Review by OkelloSam21 — COMMENTED** [review]
fine

**Comment by SaptarshiSarkar12** [comment]
> fine

Thank you for reviewing @OkelloSam21 :+1: !

**Pull request merged**